File: src/main/java/dev/rafi/vitelist/commands/sub/Remove.java
package dev.rafi.vitelist.commands.sub;

import com.velocitypowered.api.command.CommandSource;
import dev.rafi.vitelist.storage.Config;
import dev.rafi.vitelist.utils.ChatUtils;
import dev.rafi.vitelist.utils.QueryUtils;

public class Remove {
    public static void whitelistAdd(CommandSource executor, String[] args) {
        if (!executor.hasPermission("vitelist.commands.remove")) {
            ChatUtils.sendMessage(executor, Config.NO_PERMISSION);
            return;
        }

        if (args.length == 1) {
            ChatUtils.sendMessage(executor, Config.WRONG_USAGE);
        } else if (args.length == 2) {
            QueryUtils.removeFromWhiteList(args[1], "all");
            ChatUtils.sendMessage(executor, Config.REMOVE_GLOBAL.replace("%user%", args[1]));
        } else if (args.length == 3) {
            if (!QueryUtils.serverExists(args[2])) {
                ChatUtils.sendMessage(executor, Config.WRONG_SERVER.replace("%server%", args[2]));
                return;
            }
            QueryUtils.removeFromWhiteList(args[1], args[2]);
            ChatUtils.sendMessage(executor, Config.REMOVE_SERVER.replace("%user%", args[1]).replace("%server%", args[2]));
        }
    }
}

File: src/main/kotlin/dev/rafi/vitelist/commands/subcommands/Remove.kt
package dev.rafi.vitelist.commands.subcommands

import com.velocitypowered.api.command.CommandSource
import dev.rafi.vitelist.database.model.Server
import dev.rafi.vitelist.storage.Config
import dev.rafi.vitelist.utils.sendMessage
import dev.rafi.vitelist.utils.serverExists
import org.jetbrains.exposed.sql.selectAll
import org.jetbrains.exposed.sql.transactions.transaction
import org.jetbrains.exposed.sql.update

fun remove(source: CommandSource, args: Array<String>) {
    if (!(source.hasPermission("vitelist.commands.remove"))) {
        sendMessage(source, Config.NO_PERMISSION)
        return
    }
    if (args.size == 1) {
        sendMessage(source, Config.WRONG_USAGE)
    } else if (args.size == 2) {
        val playerName = args[1]

        transaction {
            Server.selectAll().forEach { server ->

                Server.update({ Server.id eq server[Server.id] }) {
                    it[players] = server[players].replace("$playerName," , "")
                }
            }
        }
        sendMessage(source, Config.REMOVE_GLOBAL.replace("%user%", playerName))
    } else {
        val playerName = args[1]
        val serverName = args[2]

        if (!serverExists(serverName)) {
            sendMessage(source, Config.SERVER_NOT_FOUND.replace("%server%", serverName))
            return
        }

        transaction {
            Server.select(Server.name, Server.players).where { Server.name eq serverName}.forEach { server ->

                Server.update({ Server.name eq server[Server.name] }) {
                    it[players] = server[players].replace("$playerName,", "")
                }
            }
        }
        sendMessage(source, Config.REMOVE_SERVER.replace("%user%", playerName).replace("%server%", serverName))
    }
}